R/make_om_part_A.R

```diff
@@ -0,0 +1,52 @@
+# Outcome measures part A
+# Establishing cohorts
+#
+
+#Is all of this related to grad rates?
+#end up with wide data (not long)
+#ask for wide data
+#maybe later do a prep script that will pivot long to wide and fix stuff
+
+#' Make Outcome Measures Part A
+#'
+#' @description Establishing the Outcome Measures cohorts
+#'
+#' @param df A dataframe of student statuses
+#' @param output A string (\code{"part"}, \code{"full"}, or \code{"both"})
+#' @param format A string (\code{"uploadable"}, \code{"readable"}, or \code{"both"})
+#'
+#' @return A text file ready for IPEDS upload
+#' @export
+#'
+#'
+make_om_part_A <- function(df, output = "part", format = "both") {
+
+  partA <- df %>%
+
+    #aggregate the full data
+    dplyr::group_by(.data$Unitid, .data$CohortType, .data$Recipient)%>%
+    dplyr::summarize(CohortCount = dplyr::n(),
+                     ExclusionCount = sum(.data$Exclusion)) %>%
+    dplyr::ungroup() %>%
+
+    #sort for easy viewing
+    dplyr::arrange(.data$CohortType, .data$Recipient) %>%
```

Comment:
IPEDS docs states "Sort Order: UNITID,SURVSECT,PART". Is that a recommendation or a requirement?

Reply:
required by IPEDS;  since we process one part at at time (in order) this is done by our script